### ISSUE TITLE
Week5 homework 질문입니다.

### ISSUE THREAD

**Issue by hwanginsun** [public-ai/ALAI_Homework#29]
[Week5_Homework3_One_VS_ALL_IrisClassifier.zip](https://github.com/public-ai/ALAI_Homework/files/3137038/Week5_Homework3_One_VS_ALL_IrisClassifier.zip)

텐서플로로 구현중에 loss값이 nan으로 표시됩니다. 
(weights와 bais도 print해보면 nan으로 표시됩니다..)

아마도 cee 함수나 다른 부분이 정상적으로 구현되지 않아 문제가 발생한 것 같습니다. 
확인부탁드립니다...

**Comment by anthony0727** [comment]
log안에 negative value가 입력 되는 것으로 추측됩니다.

1. learning rate를 조절
2. minmax scaling

으로 접근 해보시면 좋을 것 같습니다.

**Comment by hwanginsun** [comment]
감사합니다. 어제 안됬던 부분은 해결했습니다.
1. nomalization을 적용안했음
2. sigmoid 공식을 잘못써서 계산을 잘못하고 있었음...

추가적으로 현재 구현해서 학습중인데, 정확도(80%)와, 오차율(0.4)이 더이상 개선되고 있지 않습니다.
혹시 로직 또는 구현문제로 잘못한 점이 있는지 확인부탁드립니다.

이부분을 구현을 끝내고.. 클래스와 One_VS_ALL_IrisClassifier을 구현해볼 계획입니다..

![image](https://user-images.githubusercontent.com/30209848/57120951-88014600-6daf-11e9-9f15-7bea34dd2014.png)

[Week5_Homework3_One_VS_ALL_IrisClassifier.zip](https://github.com/public-ai/ALAI_Homework/files/3140728/Week5_Homework3_One_VS_ALL_IrisClassifier.zip)

**Comment by anthony0727** [comment]
일단 현재 인선님의 코드는 multi classifier입니다. 즉, 3개의 label target에 대해서 학습 하고 계십니다.

과제의 의도는, multi class를 binary class로 치환하여 진행하는 binary classification이며, 1 또는 0이라는 label값에 대해서 학습을 해야합니다.

multi classification을 할시 binary classification과 정답율, 오차율이 차이가 날 수있습니다.

아래 코드를
```python3
    def cal_acc(pred, ys) :
        if ys is 0.0 :         # Setosa
            cut_value_start = 0
            cut_value_end = 0.33
            label = tf.equal(ys, 0.0)            
        elif ys is 0.5 :       # Versicolour
            cut_value_start = 0.33
            cut_value_end = 0.66
            label = tf.equal(ys, 0.5)    
        else :                 # Virginica
            cut_value_start = 0.66
            cut_value_end = 1.01
            label = tf.equal(ys, 1.0)
```
아래 코드처럼 되돌려 주시고,
```python3
acc = tf.reduce_mean(tf.cast(tf.equal(pred_cls, self.ys), tf.float32), name='acc')
```
이와 같이 acc를 계산 하려면,
```python3
ys_data = np.where(self.ys_data == self.target, [1], [0])
``` 
와 같은식으로, label값을 치환해주면 됩니다.

예를들어, Virginica(target : 2.0)을 target label로 잡으시면,

Virginica의 label은 1,  Setosa, Versicolour의 label은 0으로 두시고 학습을 진행하시면 됩니다.

p.s.
0.33이라는 값으로 cut을 하게되면, 소수점 둘째 자리 이후의 값들은 잃어버리게 되기 때문에, 부정확한 학습이 진행 될 가능성이 있어 보입니다. 1/3, 2/3이라는 값을 쓰는것이 설득력 있어 보입니다. 0~1 range의 값 들로 학습이 진행 되기 때문에 둘째 자리 이후의 값 들도 학습에 큰 영향을 줍니다.

**Comment by hwanginsun** [comment]
답변감사드립니다 . 
multi classification문제에서 binary classfication문제로 변경하여 문제를 해결했습니다.

![image](https://user-images.githubusercontent.com/30209848/57125782-9a867a00-6dc5-11e9-8c66-5cdfb76fba8c.png)
